src/main/kotlin/com/hypto/iam/server/extensions/Validators.kt

```diff
@@ -67,7 +68,7 @@ val nameCheck = Validation<String> {
 fun <T> Validation<T>.validateAndThrowOnFailure(value: T): T {
     val result = validate(value)
     if (result is Invalid<T>) {
-        throw IllegalArgumentException(result.errors.toString())
```

Comment:
IllegalArgumentException is already converted as 400 Bad request.

https://github.com/hwslabs/iam/blob/main/src/main/kotlin/com/hypto/iam/server/ExceptionHandler.kt#L70

Fix for this issue #51 would be to add input validation in "CreateOrganization" & "CreateUser" APIs.
1. Password length (minimum 8 chars length) & Password format (1 uppercase, 1 lowercase, one special char @_#)
2. Email format validation (<username>@<domain>)
3. Phone number validation (+?<8 or more digits with support of special "-" hyphen character>)

Reply:
@hari-hypto Thank you for comments.
I tried running tests in local environment & found out that, on validation failure response status returned as 400 (which is expected in this case), however issue #51  mentioned that it is returning "unknown server error"
Please let me know incase I am missing something here.

Reply:
@shharma-vipin 
The issue here is our input validation is too lenient for password & phone number. So there were cases when the input validation was successful but exception was returned by our user management service layer (AWS Cognito) for password & phone number.

Since any failures from service layer is returned as "Internal Server Error", it was not conveyed to clients properly. So the solution is to have a more stricter validation at api layer itself so that service layer will not reject this input.

Overall the final PR version looks good to me. One minor comment is to add this validation check even for the "AdminUser" when creating organization.